server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -364,6 +364,14 @@ public DocumentDimensions getDimensions() {
 
     public abstract ContentPath path();
 
+    public final MapperBuilderContext createMapperBuilderContext() {
```

Comment:
This is not perfect, but more isolated now to the only place where it's needed. I am not entirely sure if we can remove it from here hence I removed the TODO.